cosmos/testing/integration/precompile/governance/governance_test.go

```diff
@@ -0,0 +1,185 @@
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "cosmos/testing/integration/precompile/governance:integration")
+}
+
+var (
+	tf         *integration.TestFixture
+	precompile *bindings.GovernanceModule
+)
+
+var _ = SynchronizedBeforeSuite(func() []byte {
+	// Setup the network and clients here.
+	tf = integration.NewTestFixture(GinkgoT())
+	// Setup the governance precompile.
+	precompile, _ = bindings.NewGovernanceModule(
+		common.HexToAddress("0x7b5Fe22B5446f7C62Ea27B8BD71CeF94e03f3dF2"), tf.EthClient,
+	)
+
+	return nil
+}, func(data []byte) {})
+
+var _ = SynchronizedAfterSuite(func() {
+	// Local AfterSuite actions.
+}, func() {
+	// Global AfterSuite actions.
+	os.RemoveAll("data")
+})
+
+var _ = Describe("Governance Precompile Directly", func() {
+	It("should be able to call the precompile methods directly", func() {
+		// Prepare the message.
+		govAcc := common.HexToAddress("0x7b5Fe22B5446f7C62Ea27B8BD71CeF94e03f3dF2")
+		initDeposit := sdk.NewCoins(sdk.NewInt64Coin("abera", 100))
+		message := &banktypes.MsgSend{
+			FromAddress: cosmlib.AddressToAccAddress(govAcc).String(),
+			ToAddress:   cosmlib.AddressToAccAddress(network.TestAddress).String(),
+			Amount:      initDeposit,
+		}
+		messageBz, err := message.Marshal()
+		Expect(err).ToNot(HaveOccurred())
+
+		// Prepare the Proposal.
+		proposal := v1.MsgSubmitProposal{
+			InitialDeposit: initDeposit,
+			Proposer:       cosmlib.AddressToAccAddress(network.TestAddress).String(),
+			Metadata:       "metadata",
+			Title:          "title",
+			Summary:        "summary",
+			Expedited:      false,
+		}
+		proposalBz, err := proposal.Marshal()
+		Expect(err).ToNot(HaveOccurred())
+
+		// Call the precompile create proposal method.
+		txr := tf.GenerateTransactOpts("")
+		txr.Value = big.NewInt(100)
+		tx, err := precompile.SubmitProposal(txr, proposalBz, messageBz)
+		Expect(err).ToNot(HaveOccurred())
+		ExpectMined(tf.EthClient, tx)
+
+		// Check that the proposal was created.
+		proposals, err := precompile.GetProposals(nil, 0)
+		Expect(err).ToNot(HaveOccurred())
+		Expect(proposals).To(HaveLen(2)) // genesis proposal + new proposal
+
+		// Check get proposal.
+		res, err := precompile.GetProposal(nil, 2)
+		Expect(err).ToNot(HaveOccurred())
+		Expect(res.Id).To(Equal(uint64(2)))
+
+		// Wait for the next block to be mined.
+		err = tf.Network.WaitForNextBlock()
+		Expect(err).ToNot(HaveOccurred())
+
+		// Should be able to vote on the proposal.
+		txr = tf.GenerateTransactOpts("")
+		tx, err = precompile.Vote(txr, 2, 1, "metadata")
+		Expect(err).ToNot(HaveOccurred())
+		ExpectMined(tf.EthClient, tx)
+		ExpectSuccessReceipt(tf.EthClient, tx)
+
+		// Should be able to cancel a proposal.
+		txr = tf.GenerateTransactOpts("")
+		tx, err = precompile.CancelProposal(txr, 1)
+		Expect(err).ToNot(HaveOccurred())
+		ExpectMined(tf.EthClient, tx)
+		ExpectSuccessReceipt(tf.EthClient, tx)
+	})
+
+	// @Dev - This tests fails when being run in parallel with the other tests, just rerun the cli.
```

Comment:
this one is flaky?

Reply:
stale comment. removing

Reply:
Is it something to do with ordering? 

In this case split it into another setup and tear down